src/components/AnnouncementsModal.js

```diff
@@ -0,0 +1,305 @@
+            }
+          `}
+        />
+      </Link>
+    )}
+
+    <div
+      dangerouslySetInnerHTML={{ __html: body }}
+      css={`
+        strong {
+          font-weight: bold;
+        }
+        a {
+          color: ${getValueForAnnouncementContent('linkColor')};
+        }
+        p {
+          margin-bottom: 1rem;
+        }
+        ol {
+          display: block;
+          list-style-type: decimal;
+          margin-block-start: 1em;
+          margin-block-end: 1em;
+          margin-inline-start: 0px;
+          margin-inline-end: 0px;
+          padding-inline-start: 40px;
+        }
+        ul {
+          display: block;
+          list-style-type: disc;
+          margin-block-start: 1em;
+          margin-block-end: 1em;
+          margin-inline-start: 0px;
+          margin-inline-end: 0px;
+          padding-inline-start: 40px;
+        }
+      `}
+    />
+    <Inline as="div" justifyContent="flex-end">
+      {callToAction && (
+        <Link href={callToAction}>
+          <Button>{callToActionLabel}</Button>
+        </Link>
+      )}
+    </Inline>
+  </Stack>
+);
+
+AnnouncementContent.propTypes = {
+  title: PropTypes.string,
+  imageSrc: PropTypes.string,
+  body: PropTypes.node,
+  callToAction: PropTypes.string,
+  callToActionLabel: PropTypes.string,
+};
+
+const AnnouncementsModal = ({
+  visible,
+  announcements,
+  seenAnnouncements,
+  setSeenAnnouncements,
+  onClose,
+}) => {
+  const { t } = useTranslation();
+  const cookieOptions = {
+    path: '/',
+    maxAge: 60 * 60 * 24 * 30,
+  };
+  const [cookies, setCookie] = useCookies(['seenAnnouncements']);
+  const [selectedAnnouncementId, setSelectedAnnouncementId] = useState('');
+  const [selectedAnnouncement, setSelectedAnnouncement] = useState(undefined);
+
+  const addToSeenAnnouncements = (id) => {
+    setSeenAnnouncements((prev) => (prev.length > 0 ? [...prev, id] : [id]));
+    setCookie(
+      'seenAnnouncements',
+      cookies.seenAnnouncements
+        ? [...cookies.seenAnnouncements, selectedAnnouncementId]
+        : [selectedAnnouncementId],
+      cookieOptions,
+    );
+  };
+
+  useEffect(() => {
+    if (cookies.seenAnnouncements) {
+      const updatedSeenAnnouncements = cookies.seenAnnouncements.filter(
+        (announcement) => !announcements.includes(announcement.uid),
+      );
+      setSeenAnnouncements(updatedSeenAnnouncements);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!visible || announcements.length === 0) return;
+    setSelectedAnnouncementId(announcements[0].uid);
+  }, [visible]);
+
+  useEffect(() => {
+    if (!selectedAnnouncementId) return;
+
+    const selectedAnnouncement = announcements.find(
+      (announcement) => announcement.uid === selectedAnnouncementId,
+    );
+    setSelectedAnnouncement(selectedAnnouncement);
+
+    if (!seenAnnouncements.includes(selectedAnnouncementId)) {
+      addToSeenAnnouncements(selectedAnnouncementId);
+    }
+  }, [selectedAnnouncementId]);
```

Comment:
Might make sense to reevaluate what is state and what is calculated here. I have the feeling we're going over the top with the effects here.